qiskit_nature/problems/second_quantization/lattice/lattice/line_lattice.py

```diff
@@ -0,0 +1,43 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The line lattice"""
+from .hyper_cubic_lattice import HyperCubicLattice
+
+
+class LineLattice(HyperCubicLattice):
+    """Line lattice."""
+
+    def __init__(
+        self,
+        num_nodes: int,
+        edge_parameter: complex = 1.0,
+        onsite_parameter: complex = 0.0,
+        boundary_condition: str = "open",
```

Comment:
Thanks for the suggestion. I just wasn't sure of the clear advantage of doing so, so I've left it as it was before.

Reply:
The benefit is that an Enum can be tab-completed and provides only the allowed set of available options, whereas with a plain string, typos can cause hard-to-debug scenarios. Consider e.g. the case when a user types `"opne"` instead of `"open"`. That can be very tricky to spot.
Having an Enum would allow you to write `BoundaryCondition.OPEN` (e.g.)

Reply:
Whether or not to use an Enum is a matter of preference. For example, [QC.draw](https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/circuit/quantumcircuit.py#L1663-L1682) does not use an enum, but use a string for options. It is useful to guarantee and complete the values, but there is a trade-off of more imports. In Qiskit's overall design, I think there are many cases where Enum is not used. But I'll leave it to Tamura-san. I'm fine with either.

Reply:
Thanks for the discussion.
Now, I have created an Enum `BoundaryCondition` having `BoundaryCondition.OPEN` and `BoundaryCondition.PERIODIC`.